src/components/periodic-report/dto/update-periodic-report.dto.ts

```diff
@@ -0,0 +1,44 @@
+import { Field, InputType, ObjectType } from '@nestjs/graphql';
+import { Type } from 'class-transformer';
+import { ValidateNested } from 'class-validator';
+import { CalendarDate, IdField } from '../../../common';
+import { CreateDefinedFileVersionInput } from '../../file';
+import { IPeriodicReport, PeriodicReport } from './periodic-report.dto';
+import { ReportType } from './type.enum';
+
+@InputType()
+export abstract class UpdatePeriodicReport {
+  @IdField()
+  readonly id: string;
+
+  @Field(() => ReportType)
+  readonly type: ReportType;
+
+  @Field()
+  readonly start?: CalendarDate;
+
+  @Field()
+  readonly end?: CalendarDate;
```

Comment:
These shouldn't ever change, they should be required on create and readonly from then on.

Actually I see `UploadPeriodicReport` below so this file shouldn't even be needed